src/http/request.cr

```diff
@@ -26,7 +30,9 @@ class HTTP::Request
 
   def to_io(io)
     io << @method << " " << @path << " " << @version << "\r\n"
-    HTTP.serialize_headers_and_body(io, @headers, @body)
+    cookies = @cookies
+    headers = cookies ? cookies.add_request_headers(@headers) : @headers
+    HTTP.serialize_headers_and_body(io, headers, @body)
```

Comment:
That moves the domain knowledge about how cookies are handled out of the Cookie class into Request and Response respectively though.